src/main/scala/latis/input/HapiBinaryAdapter.scala

```diff
@@ -0,0 +1,120 @@
+package latis.input
+
+import java.net.URI
+import java.nio.{ByteBuffer, ByteOrder}
+
+import cats.effect.IO
+import fs2.Stream
+import latis.data.{Data, Sample}
+import latis.model.{DataType, Function, Scalar}
+
+/**
+ * Adapter for HAPI binary datasets.
+ */
+class HapiBinaryAdapter(model: DataType) extends StreamingAdapter[Iterator[Byte]] {
+  
+  val order = ByteOrder.LITTLE_ENDIAN //all numeric values are little endian (LSB)
```

Comment:
I assume you are speaking for HAPI? I argued for big-endian which is the *network* byte order and they are streaming data over a network. The HAPI developers tend to work on little-endian PCs so they preferred little-endian. Java, created at Sun (with a network centric business model and a big-endian Solaris architecture) uses big-endian under the covers. One of these days we can migrate this to a generic BinaryAdapter and make the byte order configurable (with a default to big-endian).

Reply:
Yes, for HAPI, all numeric values are little endian.